subcommands/waves/list.go

```diff
@@ -18,15 +20,14 @@ func init() {
 	cmd.AddCommand(listCmd)
 	listCmd.Flags().Uint64P("limit", "n", 20, "Limit the number of results displayed.")
 	listCmd.Flags().IntP("page", "p", 1, "Page of waves to display when pagination is needed")
-	listCmd.Flags().BoolP("all", "a", false, "Show all waves, overrides a limit.")
 }
 
 func doListWaves(cmd *cobra.Command, args []string) {
 	factory := viper.GetString("factory")
 	limit, _ := cmd.Flags().GetUint64("limit")
 	showPage, _ := cmd.Flags().GetInt("page")
-	if showAll, _ := cmd.Flags().GetBool("all"); showAll {
-		limit = 0
+	if limit > 100500 || limit <= 0 {
+		subcommands.DieNotNil(errors.New("Limit needs to be between 1 and 100500 inclusive."))
 	}
```

Comment:
I'm not sure we need this _easter egg_ here.

In other places we only allow _specific limit values_; but here it give no benefit.
So, I'd just remove `--all` and allow any limit value.